pkg/reconcilers/base_reconciler.go

```diff
@@ -92,6 +93,41 @@ func (b *BaseReconciler) EventRecorder() record.EventRecorder {
 	return b.recorder
 }
 
+// ReconcileOptions contains options for reconcile objects.
+type ReconcileOptions struct {
+	// Ensure defaults and webhook mutations are considered before comparing
+	// existing vs desired state of resources when reconciling updates
+	// by dry-running the updates before
+	DryRunFirst *bool
+
+	GetOptions []client.GetOption
+
+	CreateOptions []client.CreateOption
+
+	DeleteOptions []client.DeleteOption
+
+	UpdateOptions []client.UpdateOption
+}
+
+// DryRunFirst sets the "dry run first" option to "true", ensuring all
+// validation, mutations etc first without persisting the change to storage.
+var DryRunFirst = dryRunFirst{}
+
+type dryRunFirst struct{}
+
+// ApplyToList applies this configuration to the given list options.
+func (dryRunFirst) ApplyToReconcile(opts *ReconcileOptions) {
+	opts.DryRunFirst = ptr.To(true)
+}
+
+var _ ReconcileOption = &dryRunFirst{}
+
+// ReconcileOption adds some customization for the reconciliation process.
+type ReconcileOption interface {
+	// ApplyToUpdate applies this configuration to the given update options.
+	ApplyToReconcile(*ReconcileOptions)
+}
+
```

Comment:
Just a matter of preference, not action needed, but usually find easier to read when types and interfaces are defined before are being used.